src/main/java/org/radarcns/webapp/resource/ManagementPortalEndPoint.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2016 King's College London and The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.radarcns.webapp.resource;
+
+import static org.radarcns.auth.authorization.Permission.Entity.SUBJECT;
+import static org.radarcns.auth.authorization.Permission.Operation.READ;
+import static org.radarcns.webapp.util.BasePath.SUBJECTS;
+import static org.radarcns.webapp.util.Parameter.PROJECT_NAME;
+import static org.radarcns.webapp.util.Parameter.SUBJECT_ID;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import java.io.IOException;
+import java.util.Collection;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.Provider;
+import org.radarcns.auth.NeedsPermission;
+import org.radarcns.auth.NeedsPermissionOnSubject;
+import org.radarcns.listener.managementportal.ManagementPortalClient;
+import org.radarcns.managementportal.Subject;
+
+/**
+ * Management Portal web-app. Function set to access subject and source information from MP. A
+ * subject is a person enrolled for in a study. A source is a device linked to the subject.
+ */
+@Provider
+@Path("/mp")
+public class ManagementPortalEndPoint {
```

Comment:
Can we rename this to SubjectEndPoint and remove the old one because all subject level information has to be taken from the Management Portal ?

Reply:
Yes, in the next PR please.